samcli/commands/build/build_context.py

```diff
@@ -307,6 +311,99 @@ def run(self):
             wrapped_from = deep_wrap if deep_wrap else ex.__class__.__name__
             raise UserException(str(ex), wrapped_from=wrapped_from) from ex
 
+    def _enable_source_maps(self):
+        """
+        Appends the --enable-source-maps NODE_OPTIONS if Sourcemap is set to true and vice versa
+        """
+        using_source_maps = False
+        incorrect_node_option = False
+
+        for stack in self.stacks:
+            for _, resource in stack.resources.items():
+                metadata = resource.get("Metadata", {})
+                if metadata.get("BuildMethod", "") != "esbuild":
+                    continue
+
+                node_option_set = self._is_node_option_set(resource)
+
+                # check if Sourcemap is provided and append --enable-source-map if not set
+                build_properties = metadata.get("BuildProperties", {})
+                source_map = build_properties.get("Sourcemap", None)
+
+                if source_map and not node_option_set:
```

Comment:
Based on the logic here it doesn't seem to have special handling to the scenario where node_option_set is true and the build_properties.get("Sourcemap") is an explicit value False, which I feel we might want to handle differently from it being None? Open to any reason for this self conflict input should not be a concern

Reply:
Yup, you're correct that it does not handle the case where NODE_OPTIONS is set and Sourcemap is false. In this case, no source map is generated, so the environment variable to enable source maps will not do anything even though it is set. The only potential concern the performance hit with source maps, I'm not quite sure if the performance is degraded with a source map file, or if its just solely by using the enable source map environment variable.

We can modify this anyways to not include that option to reduce unnecessary parameters passed. Will update this if there are no concerns with this.

Reply:
Unsure of the performance concerns here, but to @qingchm 's question is there a change in behavior from before this PR and after? If there is a conflict, we can raise that as LOG warning that there is a conflict.

Reply:
(not @qingchm answering) Before this PR, setting Sourcemaps to false would effectively disable source map usage since no map file is generated, even if the env variable was provided, this was expected behaviour. The behaviour remains the same after the PR. The `Sourcemap: false` option is basically just a way to opt out of source maps if it was defined at the Global level.